docs/src/main/asciidoc/lifecycle.adoc

```diff
@@ -110,3 +110,11 @@ When the application is stopped, the second log message is printed.
 
 As usual, the application can be packaged using `./mvnw clean package` and executed using the `-runner.jar` file.
 You can also generate the native executable using `./mvnw clean package -Pnative`.
+
+== Launch Modes
+
+Quarkus has 3 different launch modes, `NORMAL` (i.e. production), `DEVELOPMENT` and `TEST`. If you are running `quarkus:dev`
+then the mode will be `DEVELOPMENT`, if you are running a JUnit test it will be `TEST`, otherwise it will be `NORMAL`.
```

Comment:
Our Arquillian adapter supports TestNG and there the launch mode is `TEST`  too, I'm not sure how much we want to discourage people from using Arquillian, but it is possible, so perhaps we shouldn't narrow this description to just JUnit?

Reply:
At the moment we only support Arquillian for the MP TCK, this is not something we document for end users.